tools/ckb-auth-cli/src/auth_script.rs

```diff
@@ -0,0 +1,88 @@
+use auth_lock_test::AlgorithmType;
+use ckb_types::core::ScriptHashType;
+use ckb_vm::cost_model::estimate_cycles;
+use ckb_vm::registers::{A0, A7};
+use ckb_vm::{Bytes, CoreMachine, Memory, Register, SupportMachine, Syscalls};
+use hex::encode;
+use lazy_static::lazy_static;
+
+const AUTH_CODE_HASH: [u8; 32] = [0; 32];
+const AUTH_HASH_TYPE: ScriptHashType = ScriptHashType::Data1;
+
+lazy_static! {
+    pub static ref AUTH_CODE: Bytes = Bytes::from(&include_bytes!("../../../build/auth")[..]);
+}
+
+pub struct DebugSyscall {}
+impl<Mac: SupportMachine> Syscalls<Mac> for DebugSyscall {
+    fn initialize(&mut self, _machine: &mut Mac) -> Result<(), ckb_vm::error::Error> {
+        Ok(())
+    }
+
+    fn ecall(&mut self, machine: &mut Mac) -> Result<bool, ckb_vm::error::Error> {
+        let code = &machine.registers()[A7];
+        if code.to_i32() != 2177 {
+            return Ok(false);
+        }
+
+        let mut addr = machine.registers()[A0].to_u64();
+        let mut buffer = Vec::new();
+
+        loop {
+            let byte = machine
+                .memory_mut()
+                .load8(&Mac::REG::from_u64(addr))?
+                .to_u8();
+            if byte == 0 {
+                break;
+            }
+            buffer.push(byte);
+            addr += 1;
+        }
+
+        let s = String::from_utf8(buffer).unwrap();
+        println!("{:?}", s);
+
+        Ok(true)
+    }
+}
+
+pub fn run_auth_exec(algorithm_id: AlgorithmType, pubkey_hash: &[u8], message: &[u8], sign: &[u8]) {
+    let args = format!(
+        "{}:{:02X?}:{:02X?}:{}:{}:{}",
```

Comment:
The point of ckb-auth is that it can use signatures generated by other blockchain wallet and then use this signature to verify the validity of some transaction. I think here we just verify that a signature is valid, but the message of that signature is not related to any transaction. Maybe we need to generate a valid transaction, then sign a message related to that transaction and finally verify this signature is valid.

Reply:
I think ckb-auth-cli is just meant to demonstrate the availability of a certain authentication feature. If a complete test is conducted, I believe you can review the test cases (tests/cardano_lock/Makefile).

Reply:
The ckb-auth-cli will add more contracts in the future, so I created a new project for it.